src/ert/gui/model/snapshot.py

```diff
@@ -295,13 +293,12 @@ def _add_snapshot(self, snapshot: Snapshot, iter_: int):
                 NodeType.REAL,
             )
             snapshot_tree.add_child(real_node)
-            for step_id, step in real[ids.STEPS].items():
-                step_node = Node(step_id, {ids.STATUS: step[ids.STATUS]}, NodeType.STEP)
+            for step_id, step in snapshot.steps(real_id).items():
```

Comment:
When changing the way we access the metadata, why not to employ it further up in 
```python
real_id,
                {
                    ids.STATUS: real[ids.STATUS],
                    ids.ACTIVE: real[ids.ACTIVE],
                    REAL_JOB_STATUS_AGGREGATED: metadata[REAL_JOB_STATUS_AGGREGATED][
                        real_id
                    ],
                    REAL_STATUS_COLOR: metadata[REAL_STATUS_COLOR][real_id],
                },
```
and thus getting rid of `real = snapshot.data()[ids.REALS][real_id]` ?

Reply:
we did a whole refactor of the gui snapshot handling - we wanted to keep the changes minimal at first, to get a fix merged that can be used by users.

cf. #5850